packages/expect/src/utils.ts

```diff
@@ -372,6 +372,24 @@ export const partition = <T>(
   return result;
 };
 
+export const pathAsArray = (propertyPath: string): Array<any> => {
+  // will match everything that's not a dot or a bracket, and "" for consecutive dots.
+  const pattern = RegExp('[^.[\\]]+|(?=(?:\\.)(?:\\.|$))', 'g');
+  const properties: Array<string> = [];
+
+  // because the regex won't match the first dot, if present.
+  if (propertyPath[0] === '.') {
```

Comment:
I couldn't really think of a use case that would use this specific case. Why would someone have a `.` in the beginning of the path?  I've added the check, to have the an identical behavior as if we're using ```path.split('.')``` as before.

Reply:
I've figured it out while playing a match 💯. 
If such an object is checked with the path  (` .a.b.c ` ), it will find the `c` property, which's wrong. 
```JS
{a: {b: {c: {}}}}
```
I've added a negative test for this, Codecov should be happy now ✅